client/src/components/CountdownTimer.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState, useEffect, useRef } from 'react';
+
+const CountdownTimer: React.FC = () => {
+  const [timeLeft, setTimeLeft] = useState<number>(60);
+  const [isActive, setIsActive] = useState<boolean>(false);
+  const intervalRef = useRef<number | null>(null);
```

Comment:
change this to `const intervalRef = useRef<ReturnType<typeof setInterval> | null>(null);`